v2/processor.go

```diff
@@ -74,7 +103,31 @@ func NewProcessor(receiver Receiver, handler HandlerFunc, options *ProcessorOpti
 }
 
 // Start starts the processor and blocks until an error occurs or the context is canceled.
+// It will retry starting the processor based on the StartMaxAttempt and StartRetryDelayStrategy.
+// Returns the last error encountered while starting the processor.
 func (p *Processor) Start(ctx context.Context) error {
+	var savedError error
+	for attempt := 1; attempt <= p.options.StartMaxAttempt; attempt++ {
```

Comment:
Nit: Is there a reason we are starting at 1 instead of 0?

Reply:
initially put down 1 for better readability during the early exit block. changed back to start the loop from 0